packages/babel-plugin-proposal-class-properties/test/fixtures/private/logical-assignment/output.js

```diff
@@ -0,0 +1,37 @@
+function _classPrivateFieldSet(receiver, privateMap, value) { var descriptor = privateMap.get(receiver); if (!descriptor) { throw new TypeError("attempted to set private field on non-instance"); } if (descriptor.set) { descriptor.set.call(receiver, value); } else { if (!descriptor.writable) { throw new TypeError("attempted to set read only private field"); } descriptor.value = value; } return value; }
+
+function _classPrivateFieldGet(receiver, privateMap) { var descriptor = privateMap.get(receiver); if (!descriptor) { throw new TypeError("attempted to get private field on non-instance"); } if (descriptor.get) { return descriptor.get.call(receiver); } return descriptor.value; }
+
+var _nullish = new WeakMap();
+
+var _and = new WeakMap();
+
+var _or = new WeakMap();
+
+class Foo {
+  constructor() {
+    _nullish.set(this, {
+      writable: true,
+      value: 0
+    });
+
+    _and.set(this, {
+      writable: true,
+      value: 0
+    });
+
+    _or.set(this, {
+      writable: true,
+      value: 0
+    });
+  }
+
+  test() {
+    _classPrivateFieldSet(this, _nullish, _classPrivateFieldGet(this, _nullish) ?? 42);
```

Comment:
This is a fine first start, but it's [technically incorrect](https://babeljs.io/repl/build/23793/#?browsers=ie%2011&build=&builtIns=false&spec=false&loose=true&code_lz=MYGwhgzhAEBiD29oG8BQ1oHMCmAXaAxAEZgBOAFAJQrobSl4CupAdtAAwDctAvrRHkIkKAN2po60XAAtS8AO7QW2RQFFScigAMI0-IxAATFgHJ8RbNGBgQIbIa2VuGPrVzYIuKjUkyAlhAAdMRk0AD8YQC80ACMztB8rsqKCPBUge6eVJxAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-2%2Cenv&prettier=false&targets=&version=7.10.2%2Bpr.11702&externalPlugins=).

We need to perform the logical operator first, then perform the assignment in the RHS of the logical. We should test with a deep property (`this.self.#bar ??= 1`), to ensure the memoization works correctly.

Reply:
I see your point. Right now I'm clubbing these alongside `MEMBER += VALUE   ->   _set(MEMBER, _get(MEMBER) + VALUE)`. Do you propose that for logical assignment we have a different transformation `MEMBER ??= VALUE   ->   _get(MEMBER) ?? _set(MEMBER,  VALUE)`?

Reply:
I checkout out the proposal explainer (https://github.com/tc39/proposal-logical-assignment), and that's exactly what the proposal says!

Reply:
Yup, exactly.

Reply:
@jridgewell gimme a quick second.

Reply:
@jridgewell that definitely took more than a second, but it should be better now, PTAL.